File: iOS Examples/Objective-C/GranularSynthesis/GranularSynthesis/ViewController.m
//
//  ViewController.m
//  GranularSynthTest
//
//  Created by Nicholas Arner on 9/2/14.
//  Copyright (c) 2014 Aurelius Prochazka. All rights reserved.
//

#import "ViewController.h"

#import "AKFoundation.h"
#import "GranularInstrument.h"


@interface ViewController ()
{
    GranularInstrument *granularInstrument;
    BOOL isGranularInstrumentPlaying;
}

@property (strong, nonatomic) IBOutlet AKPropertySlider *mixSlider;
@property (strong, nonatomic) IBOutlet AKPropertySlider *frequencySlider;
@property (strong, nonatomic) IBOutlet AKPropertySlider *durationSlider;
@property (strong, nonatomic) IBOutlet AKPropertySlider *densitySlider;
@property (strong, nonatomic) IBOutlet AKPropertySlider *frequencyVariationSlider;
@property (strong, nonatomic) IBOutlet AKPropertySlider *frequencyVariationDistributionSlider;

@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];

    granularInstrument = [[GranularInstrument alloc] init];
    [AKOrchestra addInstrument:granularInstrument];

    _mixSlider.property       = granularInstrument.mix;
    _frequencySlider.property = granularInstrument.frequency;
    _durationSlider.property  = granularInstrument.duration;
    _densitySlider.property   = granularInstrument.density;
    _frequencyVariationSlider.property = granularInstrument.frequencyVariation;
    _frequencyVariationDistributionSlider.property = granularInstrument.frequencyDistribution;
}

- (IBAction)toggleGranularInstrument:(id)sender
{
    if (isGranularInstrumentPlaying) {
        [granularInstrument stop];
        isGranularInstrumentPlaying = NO;
    } else {
        [granularInstrument play];
        isGranularInstrumentPlaying = YES;
    }
}


@end

File: iOS Examples/Swift/GranularSynthesis/GranularSynthesis/ViewController.swift
//
//  ViewController.swift
//  GranularSynthesis
//
//  Created by Aurelius Prochazka on 2/16/15.
//  Copyright (c) 2015 Aurelius Prochazka. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    @IBOutlet var toggleSwitch: UISwitch!
    @IBOutlet var mixSlider : AKPropertySlider!
    @IBOutlet var frequencySlider : AKPropertySlider!
    @IBOutlet var durationSlider : AKPropertySlider!
    @IBOutlet var densitySlider : AKPropertySlider!
    @IBOutlet var frequencyVariationSlider : AKPropertySlider!
    @IBOutlet var frequencyVariationDistributionSlider : AKPropertySlider!
    
    let granularSynth = GranularSynth();
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        AKOrchestra.addInstrument(granularSynth)
        
        mixSlider.property                            = granularSynth.mix;
        frequencySlider.property                      = granularSynth.frequency;
        durationSlider.property                       = granularSynth.duration;
        densitySlider.property                        = granularSynth.density;
        frequencyVariationSlider.property             = granularSynth.frequencyVariation;
        frequencyVariationDistributionSlider.property = granularSynth.frequencyVariationDistribution;
    }

    @IBAction func toggleGranularInstrument(sender: AnyObject) {
        toggleSwitch.on ?  granularSynth.play() : granularSynth.stop()
    }

    
}